recipes/mpdecimal/all/conanfile.py

```diff
@@ -0,0 +1,212 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+
+class MpdecimalConan(ConanFile):
+    name = "mpdecimal"
+    description = "mpdecimal is a package for correctly-rounded arbitrary precision decimal floating point arithmetic."
+    license = "BSD-2-Clause"
+    topics = ("conan", "mpdecimal", "multiprecision", "library")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://www.bytereef.org/mpdecimal"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def configure(self):
+        if self.settings.arch not in ("x86", "x86_64"):
+            raise ConanInvalidConfiguration("Arch is unsupported")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.options.shared or self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        if self.settings.os == "Windows" and self.settings.compiler != "Visual Studio":
```

Comment:
Why didn't you use `if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
                tools.os_info.detect_windows_subsystem() != "msys2":` in this case?

Reply:
This was copied from https://github.com/bincrafters/conan-mpdecimal/blob/c8090a20020d9b6360c04117359be2a531fbd11f/conanfile.py#L36-L38
I will change it.

Reply:
winbash (msys2) is only needed when using autotools, so when the compiler != Visual Studio.
So I think this build requirement can be dropped.